packages/engine/src/world/SketchManager.ts

```diff
@@ -33,22 +43,25 @@ export class SketchManager {
   }
 
   public addSketchToScene = (instanceId: string, module: SketchModule): void => {
-    const scene = getDebugScene().scene
-    const sketch = this.createSketch(instanceId, module)
+    const engineScene = getDebugScene()
+    const scene = engineScene.scene
+    const sketch = this.createSketch(instanceId, module, engineScene)
     if (sketch.root) {
       scene.add(sketch.root)
     }
   }
 
   public removeSketchFromScene = (instanceId: string): void => {
-    const scene = getDebugScene().scene
+    const engineScene = getDebugScene()
+    const scene = engineScene.scene
     const oldSketch = scene.getObjectByName(instanceId)
 
     if (!oldSketch) {
       throw new Error(`couldn't find sketch to remove: ${instanceId}`)
     }
 
     scene.remove(oldSketch)
+    this.sketchInstances[instanceId]?.dispose?.(engineScene)
```

Comment:
Nice to have `scene` handy to clear out some environment stuff after deleting a sketch that sets it up

```ts
  dispose({ scene }) {
    scene.background = null;
    scene.environment = null;
  }
  ```

Reply:
True, sounds good, stops the sketch from needing to track internally.

Could see this causing issues with sketches that target the same thing, though that’s already the case. Maybe at some point we lock all the scene stuff down and have sketches flag themselves as a background modifier etc as to only apply the top-most. But I think that’s overkill for now, hack the planet.